AFQ/definitions/image.py

```diff
@@ -24,15 +24,23 @@ def _resample_image(image_data, dwi_data, image_affine, dwi_affine):
     Helper function
     Resamples image to dwi if necessary
     '''
-    image_type = image_data.dtype
-    if ((dwi_data is not None)
-        and (dwi_affine is not None)
-            and (dwi_data[..., 0].shape != image_data.shape)):
+    def _resample_slice(slice_data):
         return np.round(resample(
-            image_data.astype(float),
+            slice_data.astype(float),
             dwi_data[..., 0],
             image_affine,
             dwi_affine).get_fdata()).astype(image_type)
+
+    image_type = image_data.dtype
+    if ((dwi_data is not None)
+        and (dwi_affine is not None)
+            and (dwi_data.shape[:3] != image_data.shape[:3])):
```

Comment:
IIUC, this only checks the shape (and not the affine). Is there some weird case where an identically-shaped, but differently-affined image is provided?

Reply:
This is a good point. Also, it looks like the GPU image is failing for this PR... I think it might be scikit image, updating that too